Source/SA/RFmxDPD/RFmxDPD.cs

```diff
@@ -0,0 +1,202 @@
+                    ThresholdType = RFmxSpecAnMXDpdLookupTableThresholdType.Relative,
+                    ThresholdEnabled = RFmxSpecAnMXDpdLookupTableThresholdEnabled.True,
+                    ThresholdLevel = -20,
+                    LookupTableStepSize = 0.1,
+                    LookupTableCorrectionType = RFmxSpecAnMXDpdApplyDpdLookupTableCorrectionType.MagnitudeAndPhase
+
+                };
+            }
+        }
+
+        public struct MemoryPolynomialConfiguration
+        {
+            public RFmxSpecAnMXDpdApplyDpdMemoryModelCorrectionType MemoryModelCorrectionType;
+            public RFmxSpecAnMXDpdIterativeDpdEnabled IterativeDpdEnabled;
+            public int NumberOfIteration;
+            public int MemoryPolynomialOrder, MemoryPolynomialDepth, MemoryPolynomialLeadOrder, MemoryPolynomialLagOrder,
+            MemoryPolynomialLeadMemoryDepth, MemoryPolynomialLagMemoryDepth, MemoryPolynomialMaxLead, MemoryPolynomialMaxLag;
+            public static MemoryPolynomialConfiguration GetDefault()
+            {
+                return new MemoryPolynomialConfiguration
+                {
+                    MemoryModelCorrectionType = RFmxSpecAnMXDpdApplyDpdMemoryModelCorrectionType.MagnitudeAndPhase,
+                    IterativeDpdEnabled = RFmxSpecAnMXDpdIterativeDpdEnabled.False,
+                    NumberOfIteration = 3,
+                    MemoryPolynomialDepth = 2,
+                    MemoryPolynomialOrder = 3,
+                    MemoryPolynomialLeadOrder = 2,
+                    MemoryPolynomialLagOrder = 2,
+                    MemoryPolynomialLeadMemoryDepth = 2,
+                    MemoryPolynomialLagMemoryDepth = 2,
+                    MemoryPolynomialMaxLead = 2,
+                    MemoryPolynomialMaxLag = 2
+                };
+            }
+        }
+
+        public struct LookupTableResults
+        {
+            public SG.Waveform PostDPDWaveform;
+            public float[] LookupTableInputPowers;
+            public ComplexSingle[] LookupTableComplexGains;
+            public double PostDPDPAPR;
+            public double PowerOffset;
+        }
+
+        public struct MemoryPolynomialResults
+        {
+            public SG.Waveform PostDPDWaveform;
+            public ComplexSingle[] DpdPolynomial;
+            public double PostDPDPAPR;
+            public double PowerOffset;
+        }
+        #endregion
+        #region ConfigureDPD
+        public static void ConfigureCommon(RFmxSpecAnMX specAnSignal, CommonConfiguration commonConfig, Waveform waveform, string selectorString = "")
+        {
+            RFmxSpecAnMXDpdReferenceWaveformIdleDurationPresent idlePresent;
+     
+            idlePresent = waveform.IdleDurationPresent ? RFmxSpecAnMXDpdReferenceWaveformIdleDurationPresent.True : RFmxSpecAnMXDpdReferenceWaveformIdleDurationPresent.False;
+
+            specAnSignal.SelectMeasurements(selectorString, RFmxSpecAnMXMeasurementTypes.Dpd, true);
```

Comment:
@mdwhitten this is a unique case for measurement enabled since it can't be done in composite with others - just verifying this makes sense to you

Reply:
Good question. My approach for TxP power servo is to clone the existing signal so that I don’t accidentally disable previously configured measurements. In this case, however, I think that this is the only thing that we can do since this is not a self-contained function. In other words, we need to keep passing the same signal object around. 

The only thing I could think of is to create a clone of the SpecAn signal specifically for DPD in this function. Then, we retrieve that signal object by name in subsequent calls. That may avoid a slight user pain point but would probably increase confusion in the code.

Reply:
On further thought, we might want to take a rather unique approach to this use case. What if in the configure DPD function we took the RFmxInstr object in, and output a SpecAn object called `DpdSignal`. Unlike other personalities, the user won't really be managing this object themselves. Normally, the user will create the object, probably set some additional functions, and then need to call the initiate manually. In this case, Initiate is already called inside of the Apply function, which suggests to me it could be something that we create for them. This would also avoid any conflicts with existing configured SpecAn measurements, since a different signal would be used. Thoughts?

Reply:
I'm all in favor of separate signal configurations for areas where we need to use other personalities but don't want to pollute settings. I'm also in favor of creating the unique signal instance for the user and returning it to them. The only thing that makes me wary is that we choose a signal name that has very low likelyhood to cause collisions.

Reply:
@smooresni , @mdwhitten  and I discussed it. If the customer is doing another personality measurement, they would have to call RFmx specAn class to configure specAn signal, and that is independent form the personality configuration.